src/editor/Editor.js

```diff
@@ -430,6 +432,7 @@ define(function (require, exports, module) {
         // CodeMirror-focused. Instead, track focus via onFocus and onBlur
         // options and track state with this._focused
         this._focused = false;
+        collabInstance.setCodemirror(this._codeMirror);
```

Comment:
Rename `setCodemirror` to `init` and combine your logic above, removing the checks for `self.codemirror` when you handle the data events.